### PR TITLE
integrations/updater: bump dependencies

### PR DESCRIPTION
**Pull request by hugoShaka**
Bump `kube-agent-updater` dependencies.

### CODE REVIEW

**Review by marcoandredinis — COMMENTED** [review]
Some indirect packages are being downgraded :thinking: 
Do we need to upgrade compatibility to go1.20 now? Or can we wait for the release of 1.21?

**Comment by codingllama** [comment]
> Some indirect packages are being downgraded 🤔 

That is curious. Typically I don't bother much with indirect versions, barring any bugs/CVEs, but maybe revert these so we don't change those parts of the build? @hugoShaka 

> Do we need to upgrade compatibility to go1.20 now? Or can we wait for the release of 1.21?

We tend to avoid Go source-level updates for public modules like api/ - think modules that have impact code examples, automation that folks put together, etc. For "private" build modules it's your call.

File: integrations/kube-agent-updater/go.mod [hunk 1]
```diff
@@ -31,8 +31,8 @@ require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/digitorus/pkcs7 v0.0.0-20221212123742-001c36b64ec3 // indirect
 	github.com/digitorus/timestamp v0.0.0-20221019182153-ef3b63b79b31 // indirect
-	github.com/docker/cli v23.0.6+incompatible // indirect
-	github.com/docker/docker v23.0.6+incompatible // indirect
+	github.com/docker/cli v23.0.5+incompatible // indirect
```
Comment (codingllama) [inline integrations/kube-agent-updater/go.mod:34]: Revert the downgrades, so we don't change the build in those regards?

**Review by codingllama — COMMENTED** [review]
Looks reasonable, thanks Hugo!

**Comment by hugoShaka** [comment]
> Some indirect packages are being downgraded 🤔

I flushed the whole go module and re-created it with `go mod tidy`. I'll investigate

>  Do we need to upgrade compatibility to go1.20 now?

I ran with the buildbox/devbox version and the one from the main `teleport` module. I have no opinion on when we should update and to which version, I just aligned stuff (and no-one should depend on this package).

**Comment by hugoShaka** [comment]
Superseeded

**Pull request closed without merging**